File: Programmers/courses/30/lessons/42885/Solution.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0, i = 0;
        int j = people.length - 1;
        Arrays.sort(people);
        
        while (i <= j) {
            if (people[i] + people[j] <= limit) {
                i++;  
            }
            answer++;
            j--;
        }
        
        return answer;
    }
}

File: Programmers/courses/30/lessons/42885/Solution.py
def solution(people, limit):
    people.sort()
    answer = i = 0
    j = len(people) - 1
    
    while i <= j:
        if people[i] + people[j] <= limit:
            answer += 1
            i += 1
            j -= 1
        else:
            answer += 1
            j -= 1
    
    return answer

